bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/handler/RoombaHandler.java

```diff
@@ -128,7 +130,24 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     cmd = isPaused ? "resume" : "start";
                 }
 
-                sendRequest(new MQTTProtocol.CommandRequest(cmd));
+                if (cmd.startsWith(CMD_CLEAN_REGIONS)) {
+                    // format: cleanRegions:<pmid>;<region_id1>,<region_id2>,...
+                    if (Pattern.matches("cleanRegions:[^:;,]+;.+(,[^:;,]+)*", cmd)) {
+                        String[] cmds = cmd.split(":");
```

Comment:
The above pattern in no way guarantees that there is a colon to split...

Reply:
Sorry, I don't get this. Could you please give an example cmd, which would fail?

Reply:
Whoops, I just realized I'm completely in the wrong here.